File: CartyObjC/CartyObjC/RewardedVideoViewController.m
//
//  RewardedVideoViewController.m
//  CartyObjC
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

#import "RewardedVideoViewController.h"
#import <CartySDK/CartySDK.h>

@interface RewardedVideoViewController ()<CTRewardedVideoAdDelegate>

@property (nonatomic,weak)IBOutlet UILabel *infoLabel;
@property (nonatomic,strong)CTRewardedVideoAd *rewardedVideoAd;
@end

@implementation RewardedVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.rewardedVideoAd = [[CTRewardedVideoAd alloc] init];
    self.rewardedVideoAd.placementid = @"281217152650";
    self.rewardedVideoAd.delegate = self;
}

-(IBAction)loadAd:(id)sender
{
    self.infoLabel.text = @"Start load Ad";
    [self.rewardedVideoAd loadAd];
}

-(IBAction)showAd:(id)sender
{
    if(self.rewardedVideoAd.isReady)
    {
        self.infoLabel.text = @"Start show Ad";
        self.rewardedVideoAd.customRewardString = @"customRewardString";
        [self.rewardedVideoAd showAd:self];
    }
    else
    {
        self.infoLabel.text = @"Ad not Ready";
    }
}

- (void)CTRewardedVideoAdDidLoad:(nonnull CTRewardedVideoAd *)ad { 
    self.infoLabel.text = @"Ad loaded";
}

- (void)CTRewardedVideoAdDidShow:(nonnull CTRewardedVideoAd *)ad { 
    self.infoLabel.text= @"Ad did show";
}

- (void)CTRewardedVideoAdLoadFail:(nonnull CTRewardedVideoAd *)ad withError:(nonnull NSError *)error { 
    self.infoLabel.text = [NSString stringWithFormat:@"Ad load fail %@",error];
}

- (void)CTRewardedVideoAdShowFail:(nonnull CTRewardedVideoAd *)ad withError:(nonnull NSError *)error { 
    self.infoLabel.text = [NSString stringWithFormat:@"Ad show fail %@",error];
}

- (void)CTRewardedVideoAdDidClick:(nonnull CTRewardedVideoAd *)ad {
    self.infoLabel.text = @"Ad click";
}

- (void)CTRewardedVideoAdDidDismiss:(nonnull CTRewardedVideoAd *)ad {
    if([self.infoLabel.text isEqualToString:@"Ad Rewarded"])
    {
        self.infoLabel.text = @"Ad Rewarded & Ad close";
        return;
    }
    self.infoLabel.text = @" Ad close";
}

- (void)CTRewardedVideoAdDidEarnReward:(nonnull CTRewardedVideoAd *)ad rewardInfo:(nonnull NSDictionary *)rewardInfo {
    self.infoLabel.text = @"Ad Rewarded";
}

@end

File: CartySwift/CartySwift/RewardedVideoViewController.swift
//
//  RewardedVideoViewController.swift
//  CartySwift
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

import UIKit
import CartySDK

class RewardedVideoViewController: UIViewController , CTRewardedVideoAdDelegate
{
    @IBOutlet var infoLabel:UILabel?
    let rewardedVideoAd = CTRewardedVideoAd()
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        rewardedVideoAd.placementid = "281217152650"
        rewardedVideoAd.delegate = self
    }
    
    @IBAction func loadAd()
    {
        infoLabel?.text = "Start load Ad"
        rewardedVideoAd.load()
    }
    
    @IBAction func showAd()
    {
        if(rewardedVideoAd.isReady())
        {
            infoLabel?.text = "Start show Ad"
            rewardedVideoAd.customRewardString = "customRewardString"
            rewardedVideoAd.show(self)
        }
        else
        {
            infoLabel?.text = "Ad not Ready"
        }
    }
    
    func ctRewardedVideoAdDidLoad(_ ad: CTRewardedVideoAd) {
        infoLabel?.text = "Ad loaded"
    }
    
    func ctRewardedVideoAdLoadFail(_ ad: CTRewardedVideoAd, withError error: any Error) {
        infoLabel?.text = "Ad load fail \(error)"
    }
    
    func ctRewardedVideoAdDidShow(_ ad: CTRewardedVideoAd) {
        infoLabel?.text = "Ad did show"
    }
    
    func ctRewardedVideoAdShowFail(_ ad: CTRewardedVideoAd, withError error: any Error) {
        infoLabel?.text = "Ad show fail \(error)"
    }
    
    func ctRewardedVideoAdDidClick(_ ad: CTRewardedVideoAd) {
        infoLabel?.text = "Ad click"
    }
    
    func ctRewardedVideoAdDidDismiss(_ ad: CTRewardedVideoAd) {
        if(infoLabel?.text == "Ad Rewarded")
        {
            infoLabel?.text = "Ad Rewarded & Ad close"
            return;
        }
        infoLabel?.text = " Ad close"
    }
    
    func ctRewardedVideoAdDidEarnReward(_ ad: CTRewardedVideoAd, rewardInfo: [AnyHashable : Any]) {
        infoLabel?.text = "Ad Rewarded"
    }
}
